c/misra/src/rules/DIR-5-3/ThreadCreatedByThread.ql

```diff
@@ -0,0 +1,45 @@
+/**
+ * @id c/misra/thread-created-by-thread
+ * @name DIR-5-3: Threads shall not be created by other threads
+ * @description Creating threads within threads creates uncertainty in program behavior and
+ *              concurrency overhead costs.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/misra/id/dir-5-3
+ *       external/misra/c/2012/amendment4
+ *       correctness
+ *       maintainability
+ *       concurrency
+ *       performance
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+import codingstandards.cpp.Concurrency
+
+Function callers(Function f) { result = f.getACallToThisFunction().getEnclosingFunction() }
```

Comment:
There's already a standard library predicate for this - `Function.calls(Function)`.